### PR TITLE
Disposal Pipe Crafting

### PR DESCRIPTION
**Pull request by DubiousDoggo**
what it says on the tin

:cl:
- add: Added construction for disposal pipes

### CODE REVIEW

**Comment by Peptide90** [comment]
lgtm, not sure on 1 steel sheet per pipe but that's a simple thing to balance later. RPD would be something to look at in future for straight building them like an RCD.

Not sure how broken pipes are currently handled RE removing them but could be something to consider. In regards to my above comment, if broken pipes give you 1 metal sheet we don't want them to be infinitely rebuildable.

**Review by mirrorcult — CHANGES_REQUESTED** [review]
not sure if we want this even as a stopgap, since RPDs would be better. maybe a topic for the meeting

**Review by moonheart08 — APPROVED** [review]
This seems fine to me, and construction is certainly better than a pipe dispenser.

**Review by mirrorcult — APPROVED** [review]
i have decided that it is fine

**Comment by Peptide90** [comment]
> i have decided that it is fine

We happy with it only taking 1 metal sheet per pipe? In which case a destroyed pipe being removed should return no scrap.

**Review by metalgearsloth — COMMENTED** [review]
do_afters should be at least 1s IMO.

I say this because I have 350ms ping to EU and the bar essentially disappears as soon as it starts if it's 0.5s

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 1]
```diff
@@ -66,6 +66,9 @@
         !type:PhysShapeAabb
           bounds: "-0.25,-0.5,0.25,0.5"
       layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:70]: All graph IDs should be PascalCase now: DisposalPipe

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 2]
```diff
@@ -97,6 +100,9 @@
           !type:PhysShapeAabb
             bounds: "-0.25,-0.5,0.25,0.5"
         layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:104]: DisposalPipe

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 3]
```diff
@@ -124,6 +130,9 @@
           !type:PhysShapeAabb
             bounds: "-0.25,-0.5,0.25,0.4"
         layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:134]: DisposalPipe

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 4]
```diff
@@ -161,6 +170,9 @@
           !type:PhysShapeAabb
             bounds: "-0.5,-0.5,0.25,0.5"
         layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:174]: DisposalPipe

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 5]
```diff
@@ -226,6 +238,9 @@
         !type:PhysShapeAabb
           bounds: "-0.5,-0.5,0.25,0.5"
       layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:242]: DisposalPipe

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 6]
```diff
@@ -289,6 +304,9 @@
         !type:PhysShapeAabb
           bounds: "-0.5,-0.5,0.5,0.25"
       layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:308]: DisposalPipe

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml [hunk 7]
```diff
@@ -316,3 +334,6 @@
         !type:PhysShapeAabb
           bounds: "-0.5,-0.5,0.25,0.25"
       layer: [ Underplating ]
+  - type: Construction
+    graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Entities/Structures/Piping/Disposal/pipes.yml:338]: DisposalPipe

File: Resources/Prototypes/Recipes/Construction/Graphs/utilities/disposal_pipes.yml [hunk 8]
```diff
@@ -0,0 +1,133 @@
+- type: constructionGraph
+  id: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Construction/Graphs/utilities/disposal_pipes.yml:2]: DisposalPipe

File: Resources/Prototypes/Recipes/Construction/utilities.yml [hunk 9]
```diff
@@ -1,6 +1,6 @@
-#POWER
+# POWER
 - type: construction
-  name: APC
+  name: apc
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Construction/utilities.yml:3]: Acronym names to remain upper case: APC

File: Resources/Prototypes/Recipes/Construction/utilities.yml [hunk 10]
```diff
@@ -61,5 +61,95 @@
     sprite: Structures/Piping/disposal.rsi
     state: "disposal"
 
+- type: construction
+  name: disposal pipe
+  id: DisposalPipe
+  graph: disposalPipe
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Construction/utilities.yml:67]: DisposalPipe - won't tag it for every instance in this file.

**Review by Peptide90 — CHANGES_REQUESTED** [review]
Pretty much main point is PascalCase for graph IDs (I have a PR coming for everything else in recipes). Plus the APC naming change.

**Pull request merged**